### PR TITLE
[APP-4566] Tag 'latest' release with build timestamp

### PR DESCRIPTION
**Pull request by ale7714**
Use a sortable key for latest viam-server build so we can enable users to configure latest as release channel in Viam Agent config.

[APP-4566](https://viam.atlassian.net/browse/APP-4566)

[APP-4566]: https://viam.atlassian.net/browse/APP-4566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/staticbuild.yml
Comment (Otterverse) [inline .github/workflows/staticbuild.yml:79]: So I'm worried that the date here isn't going to match between different architectures. I'm not sure if that matters or not, but as arm64 and amd64 would be two different jobs, they'd run at slightly different times and thus it wouldn't be clear if/when things were the same "build" or not.

Ex: 
```
viam-server-20240410121314-arm64
viam-server-20240410121318-amd64
```

I think the correct answer would be have have a timestamp input to this workflow, so they'd both use the same one.

**Review by Otterverse — CHANGES_REQUESTED** [review]
One potential issue (inline comments.)

File: .github/workflows/staticbuild.yml
Comment (ale7714) [reply .github/workflows/staticbuild.yml:76]: ooof i see what you mean now, i'll add the new job

**Review by Otterverse — APPROVED** [review]
LGTM, but have to see if it runs correctly once you merge it. Fingers crossed.

**Pull request merged**
